File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.js
/**
 * @param {number} n
 * @return {number}
 */
var totalMoney = function(n) {
    let prevMon = 0;
    let prevDay = 1;
    let ans = 0;
    let start = 0;
    while(start < n) {
        if(start % 7 === 0) {
            prevMon++;
            ans += prevMon;
            prevDay = prevMon;
        }
        else {
            prevDay++;
            ans += prevDay;
        }
        start++;
    }
    return ans;
};

File: 1817-calculate-money-in-leetcode-bank/calculate-money-in-leetcode-bank.java
class Solution {
    public int totalMoney(int n) {
        int ans = 0;
        int start = 0;
        int prevMon = 0, prevDay = 0;
        while(start < n) {
            if(start % 7 == 0) {
                ans += ++prevMon;
                prevDay = prevMon;
            }
            else {
                ans += ++prevDay;
            }
            start++;
        }
        return ans;
    }
}